orion-rep/gmmit
# feat(pre-commit): add pre-commit hooks to improve repo code quality

### What changed?

Added pre-commit hooks configuration to the repository in order to improve code quality, security, and maintain a consistent code style.

Pre-commit hooks included:

- trailing-whitespace
- check-yaml
- check-json
- go-fmt
- go-imports
- golangci-lint
- markdownlint
- and others

### Why/Context

Automate code quality checks and ensure a consistent style.

### How to test it?

1. Install pre-commit: `brew install pre-commit`
2. Install the hooks: `pre-commit install`
3. Try to commit any changes, the hooks should run automatically.
